src/algorithms/local_search/local_search.cpp

```diff
@@ -665,6 +674,11 @@ void LocalSearch<Route,
           continue;
         }
 
+        const auto& v_t = _input.vehicles[s_t.second];
+        if (_sol[s_t.first].size() + 1 > v_t.max_number_of_tasks) {
```

Comment:
The right-hand side change is missing: we should compare to `v_s.max_number_of_tasks` where `v_s` is the source vehicle (const ref defined using `s_t.first`).